contracts/fee_splitter/src/contract.rs

```diff
@@ -0,0 +1,143 @@
+const _CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Creates a new contract with the specified parameters packed in the `msg` variable.
+///
+/// * **msg**  is message which contains the parameters used for creating the contract.
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<CoreumQueries>,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let is_weights_valid = msg
+        .addresses
+        .iter()
+        .map(|&(_, weight)| weight)
+        .fold(Decimal::zero(), |acc, x| acc + x)
+        .le(&Decimal::from_ratio(1u32, 1u32));
+
+    if !is_weights_valid {
+        return Err(ContractError::InvalidWeights {});
+    }
+
+    let config = Config {
+        addresses: msg.addresses,
+    };
+
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new().add_attribute("initialized", "contract"))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: Deps<CoreumQueries>,
+    env: Env,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg.clone() {
+        ExecuteMsg::SendTokens {
+            native_denoms,
+            cw20_addresses,
+        } => execute_send_tokens(deps, env, native_denoms, cw20_addresses),
+    }
+}
+
+fn execute_send_tokens(
+    deps: Deps<CoreumQueries>,
+    env: Env,
+    native_denoms: Vec<String>,
+    cw20_addresses: Vec<String>,
+) -> Result<Response, ContractError> {
+    let mut messages: Vec<CosmosMsg<CoreumMsg>> = vec![];
+    let config = query_config(deps)?;
+
+    native_denoms.iter().for_each(|denom| {
+        if let Ok(amount) = query_balance(&deps.querier, env.clone().contract.address, denom) {
+            for (address, decimal) in config.clone().addresses.into_iter() {
+                let send_amount = amount * decimal;
+                let msg = CosmosMsg::Bank(BankMsg::Send {
+                    to_address: address.clone().to_string(),
+                    amount: vec![Coin {
+                        denom: denom.to_string(),
+                        amount: send_amount,
+                    }],
+                });
+                messages.push(msg);
+            }
+        }
+    });
+
+    // todo - dry up and down
+
+    cw20_addresses.iter().for_each(|denom| {
+        if let Ok(amount) = query_token_balance(&deps.querier, env.clone().contract.address, denom)
+        {
+            // config.addresses.iter().for_each(|(address, decimal)| {
+            for (address, decimal) in config.clone().addresses.into_iter() {
+                let send_amount = amount * decimal;
+                let msg = CosmosMsg::Bank(BankMsg::Send {
+                    to_address: address.clone().to_string(),
+                    amount: vec![Coin {
+                        denom: denom.to_string(),
+                        amount: send_amount,
+                    }],
+                });
```

Comment:
```suggestion
                let msg = CosmosMsg::Wasm(WasmMsg::Execute {
                    contract_addr: denom.clone(),
                    msg: to_binary(&Cw20ExecuteMsg::Transfer { recipient: address.clone(), amount: send_amount })?,
                    funds: vec![],
                });
```

And for that to work you need to import `use cw20::Cw20ExecuteMsg;`

Reply:
Fixed.